### ISSUE TITLE
Try react-native-responsive-screen package

### ISSUE THREAD

**Issue by marudy** [deadcoder0904/react-native-tablets#1]
Hi @deadcoder0904, 

I have created a `development` branch locally where I have replaced usage of `react-native-size-matters` with `react-native-responsive-screen`. Can you add me as a collaborator to this repo in order to push it ?

Thanks

**Comment by deadcoder0904** [comment]
Done

**Comment by marudy** [comment]
`development` branch is pushed, have a look! I have modified all involved components.

**Comment by deadcoder0904** [comment]
Looks cool & pretty simple 🎉

But,

<img width="619" alt="rn responsive package" src="https://user-images.githubusercontent.com/16436270/39694786-12b01c66-5206-11e8-8368-4f45399ee7f9.png">

Also, since there is no diff since 2 different branches & I don't get how you did that calculations❓

So if you can provide how you calculated all those things, that would be pretty helpful

**Comment by marudy** [comment]
Looks one of the calculations is not correct :smile: Maybe something with the texts padding... Maybe padding is not needed now at all ?

Now about about calculations, I seize them with the eye, as I would do if I had a UI design without exact directions. I.e. video is about 40% of the screen's height. I put 40% and check if it is what I want it to be. If not, then I try 42% let's say.

Now about the more "strange" calculations, i.e. `font-size` I follow the same logic. I try to illustrate the provided UI design. After a few trials I see that `20DP` maps to `hp('3.75%')`. Since I know that now, I use this a reference throughout the project. So, if I have a padding of 5DP it's the 1/4 of the previous value which is 0.94% (`hp('0.94%')`). After using it for a while you become more experienced with it and you may even remember some references of values you tend to use :smiley:

**Comment by deadcoder0904** [comment]
Yeah but we need that in such a way it helps us get started quickly. The other package seems easy in this regards. 

You have an amazing library of just 2 methods, but its not coming across as easy. I think it should be explained more easily in the README. 

Or you can add a commit to `development` branch to fix it in all devices so that I can study it properly & add it to the blog.

Till then I'll be using that one. Also, please if you can make it fast since I'll be making React Native apps in a few days & I have to choose between `react-native-size-matters` & `react-native-responsive-screen` package & I know `react-native-size-matters` very well now :)

I am still not getting the % values & it looks like using % values it scales unusually & it looks a bit stretched out IMHO (regarding the video)

**Comment by marudy** [comment]
Thanks for the compliments again and the feedback. I definitely want to make the package useful and I appreciate your feedback during the whole process :)

But let's clarify something first: The basic idea is easy to understand though right? The package tries to emulate the `width: 40%` from "normal web development" to RN development as `width: wp('40%')`. That's why there is a consious choice of using the percentage symbol (%). 

Now, for some specific values, I agree it gets a bit weird when you use the % symbol for properties like `font-size` in example. But even if you were developing directly in `DP` you could still not much `px` coming from a specific design. I.e. font size of `12px` is not necessarily `12DP` (actually it almost never will be). So you need to try some DP values to see what is the correct match. The idea is instead οf trying to match `px` to `DP`, try to match px to % (and the package will automatically convert to DP). I can add this explanation to README for sure along with an example, if you think it will be helpful and more understandable.

I didn't test the result fully, so I can definitely have another look with my tablet emulators and perform a fix commit for the text chats. Regarding the video I have never used it before, so let me have another look ok?! I believe I will have the time to do it tomorrow. When do you plan to start the new project ? 

BTW, I already updated the README file(s) according to your feedback :tada:

**Comment by deadcoder0904** [comment]
Maybe this week onwards I will start making RN apps.

> But let's clarify something first: The basic idea is easy to understand though right? The package tries to emulate the width: 40% from "normal web development" to RN development as width: wp('40%'). That's why there is a consious choice of using the percentage symbol (%).

Regarding this only, I have used the `react-native-size-matters` package because it can scale based on device without flexing which both `flex` & `%` do.

The method `<size>@0.3` is the main reason it works well with that package. The main reason I have so many doubts is I didn't understand your package & I honestly don't think it should work with only those 2 methods. That's why I told you to do a PR bcz it is genius if it works for everything.

Also the example only contains colourful objects or rectangles which can easily be done just with flexbox. That's why I need to be sure if it really works for every use case bcz I will be writing about it.

Also, textual examples, videos, images like masonry layout would help. Or you can just replicate everything that is in `react-native-size-matters/examples` folder & convert it. So that way if anyone wants to see it how it can be done with `react-native-responsive-screen` rather than `react-native-size-matters` I can choose it.

Its not honestly right to compare both packages but for long now packages related to React Native Tablet Support were less & now 2 popped up. So I want to be sure which one I choose 🤔

I'll choose yours bcz its simpler but it needs to be explained better. I appreciate your work. Doing some great job here 👏

Also, you should add on what properties do you use `react-native-responsive-screen` like `font-size`, `margin`, `padding`, etc...❓

And do we need flex now that we use your package❓

And you should totally publish 1 or 2 blog posts bcz it is clearly under appreciated thing in React Native world

**Comment by marudy** [comment]
The `%` I have put on the package it's not the same one as to write `width: 40%`. In my case it is a string, it is a syntax I chose because the package is emulating the classic percentage. So we could write whatever we choose like `wp('40%')` or `wp('40')` or `wp(40)`. If you think this is confusing let's discuss it.

Now for the parts you use `react-native-responsive-screen` you don't need the normal `%` at all. Actually `react-native-responsive-screen` scales better than normal `%` in RN. Meaning that if you put `width: wp('95%')`, you will see that the end result is more consistent and better than `width: 95%`. Not sure how  RN implements the `%`, but results are not good as you mentioned yourself. Regarding Flex, I do use it along with `react-native-responsive-screen`. I would say it's not needed though and it's up to you to choose where to use it or not.

Having a look over `react-native-size-matters` examples, I am pretty confident that `react-native-responsive-screen` does all of it. In order to see what `react-native-responsive-screen` can do, please check the production app [Math Warriors](https://play.google.com/store/apps/details?id=com.wizope.mathwarriorsgame). It's entirely built with `react-native-responsive-screen`. In this app there are more than 10 views with a lot of images and animations.

For sure I can add some better explanations from what we discussed these days in project's `README`. You can use the package on any property that supports DP. I will add this information too.

Now about `react-native-tablets` app I tested with my 3 tablet emulators and I saw that `development` branch looks fine. Can you help me reproduce the problems you saw? What specifications does you tablet have?

**Nexus 7**
![screenshot from 2018-05-08 09-48-53](https://user-images.githubusercontent.com/16783589/39743128-a70333d8-52a8-11e8-93d1-fbab7a8357a8.png)

**Nexus 10**
![screenshot from 2018-05-08 09-50-32](https://user-images.githubusercontent.com/16783589/39743291-1eb527d8-52a9-11e8-8c19-ff7a35c534c5.png)

**Pixel**
![screenshot from 2018-05-08 10-14-37](https://user-images.githubusercontent.com/16783589/39743147-b198d690-52a8-11e8-85aa-ad8005ab6924.png)

**Comment by deadcoder0904** [comment]
Wow I am sold that I can use your package now. Needs some learning myself & testing it on couple of projects myself. I am using iPad Air 2 I guess. Take a look at the screen as shown in [this blog post](https://medium.com/react-native-training/scaling-react-native-apps-for-tablets-211de8399cf1).

So I am fine with using it now after I test it & this works ;)

Also, mention how you start applying scaling❓

Do you start with Mobile then go to Tablets or side-by-side Mobile & Tablets❓And list of properties on which it can be used on. I think anything that requires scaling can be used with it I guess. But you know better. I never launched scaled RN app yet.

And how did you get so many users 10,000+ so fast❓😱

**Comment by marudy** [comment]
Check out the updated `README` at [Usage](https://github.com/marudy/react-native-responsive-screen/blob/master/README.md#usage) section. I added the explanations we discussed over the previous comments.

I still need to test development branch with iPad Air 2 simulator. My guess is that some style adjustment is needed because of the iOS vs Android platform differences and not because of the `react-native-responsive-screen`. There are some minor inconsistencies between the 2 platforms, I have seen that again. It should be no big deal :) After checking/resolving that, I should be able to provide a complete PR. But I think it will take me a few days to get there, there is some catching up I need to do with other projects as well :)

I saw your blog post yes! Please go ahead and perform some development with the package and some tests of your own to become comfortable around it! :smiley:   

Regarding [Math Warriors](https://play.google.com/store/apps/details?id=com.wizope.mathwarriorsgame) I payed for some advertisement over Google and Facebook ads, so about 80 - 85% of the total downloads (~17000) has come that way. The rest 15-20% are organic downloads! I spent a lot of time performing ASO optimization over the Google Play store.

**EDIT:**
I forgot to mention that I have written [a blog post](https://medium.com/building-with-react-native/how-to-develop-responsive-uis-with-react-native-1x03-a448097c9503) explaining the responsive methods of the package. Back when I published the post, I still hadn't created the package. So maybe I can publish a new post announcing the release as you mention!

**Comment by deadcoder0904** [comment]
README has 404 links in `#2 Orientation Change` & `#3 Styled Components`

Otherwise Looks great. Thank you for doing this & I'll have to test to make it work. Yep Android & iOS are little different. Math Warriors looks great & its great you drive so much people by ads. I thought ads didn't work.

Offtopic - But can you share the numbers if its no problem❓How much ads made in the app vs how much you paid❓I am also making apps & was thinking of making it either Freemium with Ads or Just Paid. Got loads of ideas so have to try some.

**Comment by marudy** [comment]
Links were fixed and added some extra information!

Thank you for considering to try the package as well :) Math Warriors is my baby hehe, although now I'm doing something different. 

Yeah I can share numbers, I'm very open in general. I paid a total of about 1800 EUR and I got about ~13-14K installations. Then it is a different discussion how you engage these users and make them stay. It was the first time I used ads and they actually worked with a pretty good "cost per conversion". But I was doing a lot of tries and focusing to the ones that were converting bigger numbers of users and with less expenses. So I was checking ads and re-arranging things 1-2 times a day.
